class com.google.auth.oauth2.UserCredentials cannot be cast to class com.google.auth.oauth2.ServiceAccountCredentials

**Describe the bug**
IT tests fail with
```
Caused by: java.lang.ClassCastException: class com.google.auth.oauth2.UserCredentials cannot be cast to class com.google.auth.oauth2.ServiceAccountCredentials (com.google.auth.oauth2.UserCredentials and com.google.auth.oauth2.ServiceAccountCredentials are in unnamed module of loader 'app')
	at com.google.cloud.ServiceOptions$Builder.setCredentials(ServiceOptions.java:221)
```

**Sample**
```
 mvnci -Dit.datastore=true -pl spring-cloud-gcp-data-datastore
```

The issue  is a bug in `google-cloud-core:1.92.0` that is pulled from `com.google.cloud libraries-bom`. The bug was fixed in `1.92.2`.

https://github.com/googleapis/java-cloud-bom/issues/271

--- a/spring-cloud-gcp-dependencies/pom.xml
+++ b/spring-cloud-gcp-dependencies/pom.xml
@@ -24,6 +24,12 @@
 	<dependencyManagement>
 		<dependencies>
 			<dependency>
+				<!-- temporary solution for https://github.com/spring-cloud/spring-cloud-gcp/issues/2142 -->
+				<groupId>com.google.cloud</groupId>
+				<artifactId>google-cloud-core</artifactId>
+				<version>1.92.2</version>
+			</dependency>
+			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-gcp-core</artifactId>
 				<version>${project.version}</version>

